### PR TITLE
RIP github3.py

### PR DESCRIPTION
**Pull request by jhoblitt**
All usage of github3.py has been replaced by pygithub.  The later closely follows the organization of the github v3 api and is more intuitive to use.

### CODE REVIEW

File: codekit/cli/github_auth.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def main():
         while not password:
             password = getpass('Password for {0}: '.format(args.user))
 
-        note_template = '{app} via github3 on {host} by {user} {creds}'
+        note_template = '{app} via bored opossums on {host} by {user} {creds}'
```
Comment (athornton) [inline codekit/cli/github_auth.py]: [citation needed]

File: codekit/pygithub.py [hunk 2]
```diff
@@ -1,18 +1,38 @@
...
+import github
+import logging
+import textwrap
 
 logging.basicConfig()
 logger = logging.getLogger('codekit')
 
 
+class CaughtGitError(Exception):
+    """Simply exception class intended to bundle together a github.Repository
```
Comment (athornton) [inline codekit/pygithub.py]: "Simple"

File: codekit/cli/github_auth.py (hunk 1 above)
Comment (jhoblitt) [reply codekit/cli/github_auth.py]: fixed

File: codekit/pygithub.py (hunk 2 above)
Comment (jhoblitt) [reply codekit/pygithub.py]: speling haz ben fixerd

**Comment by jhoblitt** [comment]
Merged after addressing review comments.

**Pull request merged**